cumulusci/cli/cci.py

```diff
@@ -1345,30 +1343,59 @@ def task_list(runtime, plain, print_json):
 
 
 @task.command(name="doc", help="Exports RST format documentation for all tasks")
-@pass_runtime(require_project=False)
-def task_doc(runtime):
-    config_src = runtime.universal_config
-
-    click.echo("==========================================")
-    click.echo("Tasks Reference")
-    click.echo("==========================================")
-    click.echo("")
-
-    for name, options in config_src.tasks.items():
-        task_config = TaskConfig(options)
-        doc = doc_task(name, task_config)
-        click.echo(doc)
+@click.option("--project", "project", is_flag=True, help="Write project specific tasks")
+@click.option("--write", "write", is_flag=True, help="Print a json string")
+@pass_runtime(
+    require_project=False,
+)
+def task_doc(runtime, project=False, write=False):
+    if project:
+        project_config = runtime.project_config.config
+        project_name = runtime.project_config.project__name.capitalize()
+        if write:
+            Path("docs").mkdir(exist_ok=True)
+            with open("./docs/project_tasks.rst", "w", encoding="utf-8") as f:
+                f.write("==========================================\n")
+                f.write(f"{project_name} Tasks Reference\n")
+                f.write("==========================================\n")
+                f.write("\n")
+                for name, options in project_config["tasks"].items():
+                    if name in runtime.project_config.config_project["tasks"]:
+                        task_config = TaskConfig(options)
+                        doc = doc_task(name, task_config)
+                        f.write(f"{doc}")
+                        f.write("\n\n")
+        else:
+            click.echo("==========================================\n")
+            click.echo(f"{project_name} Tasks Reference\n")
+            click.echo("==========================================\n")
+            click.echo("\n")
+            for name, options in project_config["tasks"].items():
+                if name in runtime.project_config.config_project["tasks"]:
+                    task_config = TaskConfig(options)
+                    doc = doc_task(name, task_config)
+                    click.echo(f"{doc}")
+                    click.echo("\n\n")
+    else:
+        config_src = runtime.universal_config
+        click.echo("==========================================")
+        click.echo("Tasks Reference")
+        click.echo("==========================================")
         click.echo("")
 
+        for name, options in config_src.tasks.items():
+            task_config = TaskConfig(options)
+            doc = doc_task(name, task_config)
+            click.echo(doc)
+            click.echo("")
```

Comment:
Let's refactor this to avoid repeating the code that is currently repeated:
1. Determine the dict of tasks to be documented, based on the `project` option
2. Loop over the tasks, get their docs, and add to a result variable (list or string).
3. Do the actual output, by echoing the result or writing it to the file, based on the `write` option.